src/Views/Components/AdminDetailsPage/index.tsx

```diff
@@ -110,14 +113,12 @@ export default function AdminDetailsPage<T extends Record<string, any>>({
 
     // Set default values when entity is loaded
     useEffect(() => {
-        if (hasResolved) {
-            if (resetForm) {
-                resetForm(reset);
-            } else {
-                reset(record);
-            }
+        if (hasResolved && schema && record) {
+            const filteredRecord = filterOutReadOnlyFields(record, (schema as any)?.properties) as T;
+            reset(filteredRecord);
+            setIsLoading(false);
         }
-    }, [hasResolved]);
+    }, [hasResolved, schema, record]);
```

Comment:
Be careful about adding objects as a dependency in useEffect.  They cause re-renders because to comparing objects will always return false.  Instead, the workaround is to use JSON.stringify() around the object so that you are comparing a string value.